### PR TITLE
[2.5] bump rke2 version to v1.18.16+rke2r1

### PR DESCRIPTION
**Pull request by brendarearden**
**Problem**
Need to bump rke2 1.18 to latest 1.18 version

**Solution**
Bump RKE2 to v1.18.16+rke2r1 

**Issue**
https://github.com/rancher/rancher/issues/30492

### CODE REVIEW

File: cmd/rancherd/go.sum [hunk 1]
```diff
@@ -732,6 +732,8 @@ github.com/rancher/rke2 v1.18.13-0.20201116190518-b63f92a2b66e h1:0f5DPgfBH8+Y1m
 github.com/rancher/rke2 v1.18.13-0.20201116190518-b63f92a2b66e/go.mod h1:hQ5acWquDxv8rc7Nv1gXYEaZ2at4D6OwHdvlH7TxQLQ=
 github.com/rancher/rke2 v1.18.13-0.20201207193710-1d6e826b9fdf h1:YjHm/U2dMEUtYFCrLzto42IcsjVONfXoF2a7pcttIx4=
 github.com/rancher/rke2 v1.18.13-0.20201207193710-1d6e826b9fdf/go.mod h1:hQ5acWquDxv8rc7Nv1gXYEaZ2at4D6OwHdvlH7TxQLQ=
```
Comment (cbron) [inline cmd/rancherd/go.sum:734]: does tidy clean up these old versions ?

File: package/Dockerfile.runtime [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG RKE2_VERSION=v1.18.12-rke2r2
+ARG RKE2_VERSION=v1.18.16+rke2r1
```
Comment (StrongMonkey) [inline package/Dockerfile.runtime:1]: should be `-`

File: cmd/rancherd/go.sum (hunk 1 above)
Comment (StrongMonkey) [reply cmd/rancherd/go.sum:734]: I don't think so. Unless you deleted go.sum and regenerate it

File: package/Dockerfile.runtime (hunk 2 above)
Comment (brendarearden) [reply package/Dockerfile.runtime]: fixed

File: cmd/rancherd/go.sum (hunk 1 above)
Comment (brendarearden) [reply cmd/rancherd/go.sum]: I ran `go mod tidy` looks like it did get rid of the old version - pr ready for re-review

**Review by cbron — APPROVED** [review]
Works for me.

**Pull request merged**